src/toniarts/openkeeper/tools/convert/kmf/Triangle.java

```diff
@@ -23,16 +23,17 @@
  */
 public class Triangle {
 
-    private final short[] triangle;
+    private final byte[] triangle;

```

Comment:
Hmm, this doesn't break anything? We read everything one size bigger than the actual file format since Java doesn't have unsigned (well, ok, nowadays it has but those are, well, never used but looks cumbersome...).

AssetsConverter.ConvertProcess.Models I believe requires an update. This will trigger the automatic conversion.

Also if at all possible. I would prefer to have the LOD cleaning done in KmfModelLoader instead. As the files in these formats, like under package `kmf`, are supposed to be as close as possible to the original format. Raw format. Read with all, or most, of the garbage they might include. Then `KmfModelLoader `is where we interpret this data, and that is where we have to do all the hacking.

This is to try also to serve the public by keeping the file formats clear.

Good job!! Thank you so much!

Reply:
If you used the bytes directly in Java you'd get negative numbers for the larget values. But ee pass them on to the index buffers and GL supports unsigned Interpretation.

Do you always increment AssetsConverter versions to avoid confusion about code vs cache or only when it's a breaking change to reduce the number of reprocessings?

Good point about the loader, will do.

Reply:
It is a good practice to increase it always when a loader changes. From here I can't determine is there an actual change in the models or not. Then, like you said, code matches the _cache_. Also good to monitor that nothing breaks and if the file sizes suddenly increase drastically... is there going to be a performance hit etc.

This is development stage. I wouldn't worry about the asset conversion process. I have Skylake era Intel processor and all that, it is just 2 minutes to do the complete conversion. And increasing just the model version number will just trigger model conversion. One time.